packages/cli/lib/lib/webpack/webpack-sw-config.js

```diff
@@ -0,0 +1,62 @@
+const webpack = require('webpack');
+const merge = require('webpack-merge');
+const baseConfig = require('./webpack-base-config');
+const BabelEsmPlugin = require('babel-esm-plugin');
+const fs = require('fs');
+const {yellow} = require('chalk');
+const { resolve } = require('path');
+
+function swConfig(config) {
+  const { dest, src, brotli, esm } = config;
+
+  const plugins = [
+    new webpack.DefinePlugin({
+      'process.env.ES_BUILD': false,
+      'process.env.ENABLE_BROTLI': brotli,
+      'process.env.NODE_ENV': 'production',
+    }),
+  ];
+
+  esm && plugins.push(
+    new BabelEsmPlugin({
+      filename: '[name]-esm.js',
+      beforeStartExecution: (plugins) => {
+        plugins.forEach(plugin => {
+          if (plugin.constructor.name === 'DefinePlugin' && plugin.definitions) {
```

Comment:
What's the reason for only mutating values in-place here - does this fix `DefinePlugin` definitions leaking into the parent compiler?

Otherwise this could just be

```js
plugins.forEach(plugin => {
  if (plugin.constructor.name === 'DefinePlugin') {
    if (!plugin.definitions) throw Error('ESM Error:  DefinePlugin found without definitions.');
    plugin.definitions['process.env.ES_BUILD'] = 'true';
  }
});
```

Reply:
good point, i can chk that

Reply:
works, fixed